pkg/controller/control.go

```diff
@@ -213,6 +213,14 @@ func (c *Controller) Expand(size int64) error {
 	go func(size int64) {
 		expanded := false
 		defer func() {
+			// Frontend expansion involves in the iSCSI session rescanning, which will wait for the in-fly io requests complete.
+			// Hence there will be a deadlock once we use the lock to protect this frontend expansion.
```

Comment:
Could you please explain more about how a deadlock would happen here? Thank you

Reply:
1. The online expansion flow adds a lock for this frontend expansion.
1. New r/w requests will be blocked once reaching the controller.
2. The frontend expansion will try to rescan/refresh the iSCSI initiator, which actually needs to flush and wait for all in-fly r/w requests. But there are some requests getting stuck (in step 2), deadlock

Reply:
Ah. Do you mean if we do [this](https://github.com/shuo-wu/longhorn-engine/blob/7376a90ceeb70caf205e10352c40dae054dd5dae/pkg/controller/control.go#L239-L240) before [this](https://github.com/shuo-wu/longhorn-engine/blob/7376a90ceeb70caf205e10352c40dae054dd5dae/pkg/controller/control.go#L215), the deadlock will happen?

Reply:
Thank you for the explantion!